_posts/2020-03-30-exploting-spatial-diversity-using-unetstack3.md

```diff
@@ -0,0 +1,98 @@
+![Overview](../assets/img/sd.png)
+
+An illustration of the general overview of such a receiver system is shown above and few terminologies might help for further discussion:
+
+- *Transmitter:* A node which transmits information
+- *Main receiver:* A node that acts as the main receiver in a group of receiver nodes. It is the main receiver's responsibility to decode the information.
+- *Assisting receiver:* A node that acts as an assisting receiver in a group of receiver nodes. It is the assisting receiver's responsibility to forward the relevant information to the main receiver.
+
+The two assisting receiver nodes are in cahoots with the main receiver to cooperatively share the information! This sharing of information usually happens over a short-range wired or wireless network (e.g., WiFi, TCP/IP, UDP/IP). Although, nothing stops one from using a different technology for sharing information.
+
+### What's the immediate practical advantage one can see ?
+
+Consider a case where you are on a ship with a modem deployed and you've been receiving status updates from a deployed AUV every few minutes. The AUV moves into an area where the connectivity is poor, and you can no longer successfully receive the status reports. We've all experienced being in that situation where the underwater communication link breaks, haven't we?
+
+If you have a second modem available on the ship, you deploy it from another part of the ship. Or maybe there is a gateway buoy deployed nearby with a modem, and you can connect to its modem. Either way, the distributed spatial diversity technique (`Unity`) magically uses the information from both modems to recover connectivity to the AUV! 
+
+### So how to use `Unity` in UnetStack 3.1?
+
+The `Unity` is implemented as a premium agent and is available with the latest UnetStack 3.1 release. It allows users to transparently implement spatial diversity with a set of COTS UnetStack-based underwater acoustic modems, each with only a single receiver.
+
+Configuring and using the `Unity` agent to exploit spatial diversity is easy with just two simple steps:
+
+1. Setup the receiver nodes to cooperate.
+2. Add the `Unity` agent on the main receiver.
+
+
+#### 1. Set up for receiver nodes to cooperate (an example): 
+
+To set up the group of receivers to cooperate over a short-range network, we use [`Wormhole`](https://unetstack.net/handbook/unet-handbook_preface.html)  agent provided in the latest release UnetStack 3.1.  Transmitter node makes a transmission that is heard at all the receiver nodes. However, none of the nodes are able to successfully recover the information received, as the communication link is noisy.  In order to share the received noisy signals among the receivers, we can connect the receiver nodes using a `Wormhole`. A connection over a UDP link over a WiFi network is established by adding just a few lines of code on the receiver nodes as shown below:
+
+```groovy
+container.add 'udp', new UdpLink()            (1)
```

Comment:
People may try typing `(1)` in the console, or copy and paste the whole line. If you want to tag a line, you should use valid code, i.e., but the tag in a comment: `// (1)`.

Reply:
Resolved by @notthetup's fixes.